### PR TITLE
Feature: Add CSS variations to lesson note boxes #3431

### LINKED ISSUE #3431: Feature: Different Note Boxes in Lesson Content
_Originally by @KevinMulhern_

### Why?
We are already using a standard note box which draws attention to important information https://www.theodinproject.com/lessons/foundations-installations#introduction

We could have different note boxes with different styles to give the information more context. For example
Tip - green icon/background
Warning - red icon/background
note - yellow or blue icon/background

### Examples
[JS Info](https://javascript.info/strict-mode)

Warning box:
Image

Note Box
Image

### PR DESCRIPTION
**Pull request by luuu-xu**
## Because

Different variations (note, extra, warning) of lesson note boxes draw attention to readers for important/warning information.

## This PR

- Added CSS classes to varied lesson note boxes, including the use of corresponding icon by Fontawesome.

## Issue

Closes #3431 

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/theodinproject/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `keyword: brief description of change` format, using one of the following keywords:
  - `Feature` - adds new or amends existing user-facing behavior
  - `Chore` - changes that have no user-facing value, refactors, dependency bumps, etc
  - `Fix` - bug fixes
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] I have verified all tests and linters pass after making these changes.
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [ ] If applicable, this PR includes new or updated automated tests

### CODE REVIEW

File: app/javascript/stylesheets/lesson-content.css [hunk 1]
```diff
@@ -27,6 +27,38 @@
...
+  .lesson-note > h4::before {
+    font-family: "Font Awesome 6 Free";
+    font-weight: 900;
+    font-size: large;
+    content: "\f249";
+    @apply mr-2 text-gold dark:text-gold-700;
+  }
+
+  .lesson-note--extra {
+    @apply border-purple-600 dark:border-purple-400;
```
Comment (KevinMulhern) [inline app/javascript/stylesheets/lesson-content.css:39]: If we change this to tip or something like that, green would be a more appropriate color.

```suggestion
    @apply border-green-600 dark:border-green-700;
```

File: app/javascript/stylesheets/lesson-content.css [hunk 2]
```diff
@@ -27,6 +27,38 @@
...
+  .lesson-note--extra > h4::before {
+    font-family: "Font Awesome 6 Free";
+    font-weight: 900;
+    font-size: large;
+    content: "\f0eb";
+    @apply mr-2 text-purple-600 dark:text-purple-400;
+  }
+
+  .lesson-note--warning {
+    @apply border-red-600 dark:border-red-400;
```
Comment (KevinMulhern) [inline app/javascript/stylesheets/lesson-content.css:51]: Nit: red-400 looks a little washed out to me on dark mode, red-500 has a better feel. Lets also up the light mode color to red-700 for a little bit of better contrast:

```suggestion
    @apply border-red-700 dark:border-red-500;
```

Screenshot 2023-06-07 at 10 12 45

Screenshot 2023-06-07 at 10 12 28

**Review by KevinMulhern — CHANGES_REQUESTED** [review]
Thanks for all your work on these 💪 just a few minor suggestions

Because our markdown is pulled in from an outside source, we'll also need to add `lesson-note--tip` and `lesson-note--warning` to the tailwind safe list [here](https://github.com/TheOdinProject/theodinproject/blob/main/tailwind.config.js#L13) to get these new classes to work.

File: app/javascript/stylesheets/lesson-content.css (hunk 2 above)
Comment (luuu-xu) [reply app/javascript/stylesheets/lesson-content.css:51]: Looks nice to me!

File: app/javascript/stylesheets/lesson-content.css (hunk 1 above)
Comment (luuu-xu) [reply app/javascript/stylesheets/lesson-content.css:39]: Yes, I agree lesson-note--tip would be better with a green colour.

**Review by KevinMulhern — APPROVED** [review]
They look great @luuu-xu 💪  thanks for doing this

**Comment by luuu-xu** [comment]
Thank you @KevinMulhern for all the help! That was a great first PR for me and I will definitely contribute more to TOP!!

**Pull request merged**